### PR TITLE
Page Object Model for both Tests

### PR DESCRIPTION
**Pull request by Millionhat**

### CODE REVIEW

File: src/main/java/pages/GooglePage.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Object class for the google webpage.
+ */
+public class GooglePage {
+
+  public void goToGoogle(WebDriver driver) {
+    driver.get("https://www.google.com/");
+  }
+
+  public void googleSearch(WebDriver driver, String search) {
```
Comment (aperdomob) [inline src/main/java/pages/GooglePage.java]: change the name only to search

File: src/main/java/pages/GooglePage.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    driver.findElement(By.name("q")).clear();
```
Comment (aperdomob) [inline src/main/java/pages/GooglePage.java]: `By.name("q")` should be a variable

File: src/main/java/pages/SaucedomoCatalogPage.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * Object class for the Catalog page.
+ */
+public class SaucedomoCatalogPage {
+
+  public void selectItem(WebDriver driver) {
+    driver.findElement(By.id("add-to-cart-sauce-labs-bolt-t-shirt")).click();
+  }
+
+  public void goToShoppingCart(WebDriver driver) {
+    driver.findElement(By.id("shopping_cart_container")).click();
```
Comment (aperdomob) [inline src/main/java/pages/SaucedomoCatalogPage.java]: the driver should be pass in the constructor

File: src/main/java/pages/SaucedomoCheckOutPage.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    driver.findElement(By.id("last-name")).sendKeys("Palma");
+  }
+
+  public void fillOutPostalCode(WebDriver driver) {
+    driver.findElement(By.id("postal-code")).sendKeys("77901");
+  }
+
+  public void proceedToVerification(WebDriver driver) {
+    driver.findElement(By.id("continue")).click();
+  }
```
Comment (aperdomob) [inline src/main/java/pages/SaucedomoCheckOutPage.java]: should have one method to do all steps, also the parameters should be passed like input parameters

File: src/main/java/pages/SaucedomoLogInPage.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    driver.findElement(By.id("user-name")).sendKeys("standard_user");
+  }
+
+  public void fillPassword(WebDriver driver) {
+    driver.findElement(By.id("password")).sendKeys("secret_sauce");
+  }
+
+  public void submitInfo(WebDriver driver) {
+    driver.findElement(By.id("login-button")).click();
+  }
```
Comment (aperdomob) [inline src/main/java/pages/SaucedomoLogInPage.java]: join in a only method and the parameter should set with input parameters

File: src/test/java/BuyTshirtProcessTest.java [hunk 6]
```diff
@@ -36,18 +46,23 @@ public void setup() {
...
+    checkOut.fillOutName(driver);
+    checkOut.fillOutLastName(driver);
+    checkOut.fillOutPostalCode(driver);
+    checkOut.proceedToVerification(driver);
+
+    verification.finishProcess(driver);
+
     Assert.assertNotNull(driver.findElement(By.id("checkout_complete_container")));
     Assert.assertTrue(driver.findElement(By.id("checkout_complete_container"))
         .findElement(By.className("complete-header")).getText().contains("THANK YOU"));
```
Comment (aperdomob) [inline src/test/java/BuyTshirtProcessTest.java]: the test should not have locators

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: src/main/java/pages/GooglePage.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+
+/**
+ * Object class for the google webpage.
+ */
+public class GooglePage {
+
+  private String location = "q";
```
Comment (aperdomob) [inline src/main/java/pages/GooglePage.java]: the idea is store the selenium locator instead of string

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: src/test/java/BuyTshirtProcessTest.java [hunk 8]
```diff
@@ -31,26 +41,33 @@ public void setup() {
...
+
+    shoppingCart.goToCheckOut();
+
+    checkOut.fillOutForm("Juan", "Palma", "77601");
+
+    verification.finishProcess();
+
+    Object container = driver.findElement(By.id("checkout_complete_container"));
+    String result = driver.findElement(By.id("checkout_complete_container"))
+        .findElement(By.className("complete-header")).getText();
```
Comment (aperdomob) [inline src/test/java/BuyTshirtProcessTest.java]: Into a test you should not call locators, I mean, all locators should encapsulated into page object models

File: src/test/java/GoogleSearchTest.java [hunk 9]
```diff
@@ -33,21 +36,21 @@ public void setup() {
...
+    google.goToGoogle();
+
+    google.search("Mjolnir");
     Reporter.log("First search task completed");
-    driver.findElement(By.name("q")).clear();
-    driver.findElement(By.name("q")).sendKeys("perficient"
-        + Keys.RETURN);
+
+    google.search("perficient");
     Reporter.log("Second search task completed");
```
Comment (aperdomob) [inline src/test/java/GoogleSearchTest.java]: can you explain this line?

File: src/test/java/GoogleSearchTest.java [hunk 10]
```diff
@@ -33,21 +36,21 @@ public void setup() {
...
-    driver.findElement(By.id("rso")).findElements(By.xpath("/*"))
-      .get(0).findElement(By.tagName("h3")).click();
+
     Assert.assertTrue(driver.getTitle().contains("perficient"));
```
Comment (aperdomob) [inline src/test/java/GoogleSearchTest.java]: Into a test you should not call selenium methods, I mean, all calls should be encapsulated into page object models

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: src/test/java/GoogleSearchTest.java (hunk 9 above)
Comment (Millionhat) [reply src/test/java/GoogleSearchTest.java]: I had two different searches just to play around with the commands, it will be removed

**Review by aperdomob — APPROVED** [review]

**Pull request merged**